### PR TITLE
RIA-3451 CO and Judge decide an application

### PR DESCRIPTION
**Pull request by tarun-palisetty**
### JIRA link (if applicable) ###
[RIA-3457](https://tools.hmcts.net/jira/browse/RIA-3457)

### Change description ###
CaseOfficer and Judge decide an online application

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by Lee365 — CHANGES_REQUESTED** [review]
As discussed, try to removed sonar code errors of bugs and code smells.

**Review by Lee365 — APPROVED** [review]
LGTM

**Review by Lee365 — DISMISSED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/FeatureTogglerService.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+public class FeatureTogglerService {
+
+    private FeatureToggler featureToggler;
+
+    public FeatureTogglerService(FeatureToggler featureToggler) {
+        this.featureToggler = featureToggler;
+    }
+
+    public boolean getValueForMakeAnApplicationFeature() {
+        return featureToggler.getValue("make-an-application-feature", false);
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/FeatureTogglerService.java:15]: We do not need separate service to wrap only one line of code, please remove and use inline `featureToggler.getValue("make-an-application-feature", false)` in handlers

**Review by makowalski — CHANGES_REQUESTED** [review]
rebasing on master and some code style changes needed

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/FeatureTogglerService.java (hunk 1 above)
Comment (tarun-palisetty) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/FeatureTogglerService.java]: Now the FeatureToggler call is made inline in the handler.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DecideAnApplicationHandler.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                makeAnApplication.setDecisionDate(dateProvider.now().toString());
+                makeAnApplication.setDecisionMaker(decisionMakerRole);
+                asylumCase.write(HAS_APPLICATIONS_TO_DECIDE, YesOrNo.NO);
+            });
+
+        mayBeMakeAnApplications
+            .orElse(Collections.emptyList())
+            .stream()
+            .filter(a -> a.getValue().getDecision().equals("Pending"))
+            .findAny().ifPresent(a -> asylumCase.write(HAS_APPLICATIONS_TO_DECIDE, YesOrNo.YES));
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DecideAnApplicationHandler.java:92]: :+1:

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**